README.md

```diff
@@ -105,7 +104,31 @@ However, be aware that phantomjs process will get detached (and thus won't exit)
 
 ## `page` object API
 
-  The `page` object that is returned with `#createPage` is a proxy that sends all methods to `phantom`. Most method calls should be identical to PhantomJS API. You must remember that each method returns a `Promise`.
+The `page` object that is returned with `#createPage` is a proxy that sends all methods to `phantom`. Most method calls should be identical to PhantomJS API. You must remember that each method returns a `Promise`.
+
+For list of available methods, properties and events see [PhantomJS documentation of WebPage module](http://phantomjs.org/api/webpage/).
+
+### `page#loadContent`
+
+`page.loadContent` is used to set content and/or URL address of the page.
+
+It differs from `page.invokeMethod('setContent')` by waiting for contents to fully load.
+
+It returns `Promise` resolving after `onLoadFinished` (handled internally), thus providing workaround for [PhantomJS issue #12750](https://github.com/ariya/phantomjs/issues/12750).
+
+The returned `Promise` can yield `'success'` or `'fail'`, as per [PhantomJS WebPage documentation](http://phantomjs.org/api/webpage/handler/on-load-finished.html), or `false` in case of IPC error.
+
+```js
+let htmlContent = '<html><head><title>setContent Title</title></head><body></body></html>';
+let pageLocation = 'http://localhost:8888/';
+page.setContent(htmlContent, pageLocation).then(function (result) {
```

Comment:
> First we've overwritten setContent but then decided to revert it and add loadContent separately.

Why? Couldn't you just implement what you did in shim.js with node? Do setContent and add the right loaders. That would work.